components/compose/cfr/src/main/java/mozilla/components/compose/cfr/helper/ScreenOrientation.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.compose.cfr.helper
+
+fun interface ScreenOrientationListener {
+    fun onScreenOrientationChanged(orientation: Orientation)
+}
+
+enum class Orientation
+
+// WIP
+object ScreenOrientation {
```

Comment:
Fenix uses `GeckoScreenOrientation` from `geckoview` which seems like a just a bit too big of a dependency to add for just that listener.
So we have to add our own.
Thinking that it might be useful to have it in a separate module, available to other ones which don't need the CFR.

Reply:
I saw this earlier when investigating other CFRs for Focus and I was curious, why do we use the GeckoScreenOrientation listener for the CFRs? It seems like it's used for knowing when the app was rotated and we can get that directly from Android APIs.

Reply:
+100 for this upstream change though. :)

Reply:
Was using `GeckoScreenOrientation` since it was already available and ready to use.
Having to find a replacement now reminded me that although there are Android APIs for this (which `GeckoScreenOrientation` uses), there are many such APIs, some deprecated in alter versions of Android and there is simple listener that would just work.
For example the [OrientationEventListener](https://developer.android.com/reference/android/view/OrientationEventListener) Google recommends will also fire if "Auto-rotate" is disabled, similar to a sensor listener though the display will not actually be rotated.